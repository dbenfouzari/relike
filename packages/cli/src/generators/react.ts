import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
