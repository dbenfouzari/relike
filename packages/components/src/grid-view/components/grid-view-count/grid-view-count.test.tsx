import { render } from "@testing-library/react";

import Axis from "../../../axis";
import Colors from "../../../colors";
import GridViewCount from "./grid-view-count";

const items = Array.from({ length: 32 }, (_, index) => {
  const color = Colors.primaries[Math.floor(Math.random() * Colors.primaries.length)];

  return (
    <div key={index} style={{ backgroundColor: color["400"].toRGBA() }}>
      Item {index}
    </div>
  );
});

describe("GridView.Count", () => {
  it("should render successfully", () => {
    render(<GridViewCount crossAxisCount={4}>{items}</GridViewCount>);
  });

  it("should render successfully with horizontal scroll", () => {
    const consoleWarnMock = jest.spyOn(console, "warn").mockImplementation();

    render(
      <GridViewCount scrollDirection={Axis.horizontal} crossAxisCount={4}>
        {items}
      </GridViewCount>,
    );

    expect(consoleWarnMock).toHaveBeenCalledWith(
      "[GridView.Count] You used Axis.horizontal as scrollDirection. Be warned that strange behaviour may occur with this.",
    );
  });
});
