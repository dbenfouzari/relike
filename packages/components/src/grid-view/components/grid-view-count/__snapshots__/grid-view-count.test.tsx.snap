// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridView.Count should render successfully 1`] = `
<div
  class="wrapper-0-2-1 wrapper-d0-0-2-3"
  data-testid="grid-view-count"
>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    0
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    1
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    2
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    3
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    4
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    5
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    6
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    7
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    8
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    9
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    10
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    11
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    12
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    13
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    14
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    15
  </div>
</div>
`;

exports[`GridView.Count should render successfully with horizontal scroll 1`] = `
<div
  class="wrapper-0-2-1 wrapper-d1-0-2-4"
  data-testid="grid-view-count"
>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    0
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    1
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    2
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    3
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    4
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    5
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    6
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    7
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    8
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    9
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    10
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    11
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    12
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    13
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    14
  </div>
  <div
    class="child-0-2-2"
    style="background-color: rgb(66, 165, 245);"
  >
    Item 
    15
  </div>
</div>
`;
